zerver/views/streams.py

```diff
@@ -186,6 +186,8 @@ def remove_subscriptions_backend(request, user_profile,
     (removed, not_subscribed) = bulk_remove_subscriptions(people_to_unsub, streams)
 
     for (subscriber, stream) in removed:
+        if stream.invite_only and stream.num_subscribers() == 0:
+            do_deactivate_stream(stream)
```

Comment:
This sort of logic should be moved down to `bulk_remove_subscriptions`, I think (since subscriptions are removed on other code paths as well, but all of those paths call `bulk_remove_subscriptions`), and maybe also a test should be added.